projects/angular/src/forms/datepicker/date-input.validator.ts

```diff
@@ -33,3 +34,57 @@ export class ClrDateInputValidator implements Validator {
     return null;
   }
 }
+
+@Directive({
+  selector: '[clrStartDate]',
+  providers: [{ provide: NG_VALIDATORS, useExisting: ClrStartDateInputValidator, multi: true }],
+})
+export class ClrStartDateInputValidator implements Validator {
+  constructor(
+    @Optional() private dateIOService: DateIOService,
+    @Optional() private dateNavigationService: DateNavigationService
+  ) {}
+
+  validate(control: AbstractControl): ValidationErrors {
+    if (this.dateIOService) {
+      const value = this.dateIOService.getDateValueFromDateString(control.value);
+      if (
+        value &&
+        this.dateNavigationService &&
+        this.dateNavigationService.selectedEndDay &&
+        value > this.dateNavigationService.selectedEndDay.toDate()
+      ) {
+        return { range: { startDate: value, endDate: this.dateNavigationService.selectedEndDay.toDate() } };
+      }
+    }
+
+    return null;
+  }
+}
+
+@Directive({
+  selector: '[clrEndDate]',
+  providers: [{ provide: NG_VALIDATORS, useExisting: ClrEndDateInputValidator, multi: true }],
+})
+export class ClrEndDateInputValidator implements Validator {
+  constructor(
+    @Optional() private dateIOService: DateIOService,
+    @Optional() private dateNavigationService: DateNavigationService
+  ) {}
+
+  validate(control: AbstractControl): ValidationErrors {
+    if (this.dateIOService) {
+      const value = this.dateIOService.getDateValueFromDateString(control.value);
+      if (
+        value &&
+        this.dateNavigationService &&
+        this.dateNavigationService.selectedDay &&
+        value < this.dateNavigationService.selectedDay.toDate()
+      ) {
+        return { range: { startDate: this.dateNavigationService.selectedDay.toDate(), endDate: value } };
+      }
```

Comment:
```suggestion
      const value = this.dateIOService.getDateValueFromDateString(control.value);
      const startDate = this.dateNavigationService?.selectedDay?.toDate();

      if (value && startDate && value < startDate) {
        return { range: { startDate, endDate: value } };
      }
```

This code can be simplified.